src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_binary_call_emitter.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_binary_call_emitter.hpp"
+
+#include <xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_reg.h>
+
+#include <algorithm>
+#include <cpu/aarch64/cpu_isa_traits.hpp>
+#include <cpu/aarch64/jit_generator.hpp>
+#include <cstddef>
+#include <set>
+#include <utility>
+#include <vector>
+
+#include "emitters/utils.hpp"
+#include "snippets/emitter.hpp"
+#include "utils/general_utils.h"
+
+using namespace dnnl::impl::cpu::aarch64;
+
+namespace ov::intel_cpu::aarch64 {
+
+jit_binary_call_emitter::jit_binary_call_emitter(jit_generator* h, cpu_isa_t isa, std::set<snippets::Reg> live_regs)
+    : jit_emitter(h, isa),
+      m_regs_to_spill(std::move(live_regs)) {}
+
+const std::set<snippets::Reg>& jit_binary_call_emitter::get_regs_to_spill() const {
+    OV_CPU_JIT_EMITTER_ASSERT(m_regs_initialized, "Binary call registers must be initialized first");
+    return m_regs_to_spill;
+}
+
+const Xbyak_aarch64::XReg& jit_binary_call_emitter::get_call_address_reg() const {
+    OV_CPU_JIT_EMITTER_ASSERT(m_regs_initialized, "Binary call registers must be initialized first");
+    return m_call_address_reg;
+}
+
+const Xbyak_aarch64::XReg& jit_binary_call_emitter::get_callee_saved_reg() const {
+    OV_CPU_JIT_EMITTER_ASSERT(m_regs_initialized, "Binary call registers must be initialized first");
+    return m_callee_saved_reg;
+}
+
+void jit_binary_call_emitter::init_binary_call_regs(size_t num_binary_args,
+                                                    const std::vector<size_t>& used_gpr_idxs) const {
+    if (m_regs_initialized) {
+        return;
+    }
+
+    // ARM64 AAPCS: X0-X7 are parameter/result registers, X8-X18 are temporary registers
+    // X19-X28 are callee-saved registers, X29 is frame pointer, X30 is link register, X31 is stack pointer
+
+    std::vector<size_t> all_used_idxs = used_gpr_idxs;
+
+    // Add ABI parameter registers to used list (X0-X7 based on num_binary_args)
+    for (size_t i = 0; i < std::min(num_binary_args, static_cast<size_t>(8)); i++) {
```

Comment:
Should we throw exception if `num_binary_args > 8`? Because we cannot provide such count of abi params

Reply:
Done